orion/main.go

```diff
@@ -59,7 +60,14 @@ func main() {
 		apiAchieve.POST("/v1/achievements/:id", achieve.UpdateAchievement)
 		apiAchieve.DELETE("/v1/achievements/:id", achieve.DeleteAchievement)
 	}
-	// apiSemesters := router.Group("api/semesters/")
+	apiSemesters := router.Group("api/semesters/")
+	{
+		apiSemesters.GET("/v1/all", semesters.GetSemesters)
+		apiSemesters.POST("/v1/create", semesters.CreateSemester)
+		apiSemesters.GET("/v1/semesters/:id", semesters.GetSemester)
```

Comment:
The group should stay as `/api/semesters/`
The GET, POST and DELETE here should indicate we are operating on one `semester`.
So can you change the following to `v1/semester/:id`